File: iOS/ColoringBookTests/MockImageService.swift
import Foundation
@testable import ColoringBook

actor MockImageService: ImageServiceProtocol {
    var fetchImagesResult: Result<[ColoringImage], Error> = .success([])
    var generateImageResult: Result<ColoringImage, Error> = .success(
        ColoringImage(id: "gen1", filename: "gen.png", title: "Generated", prompt: "test", url: "/images/gen.png", thumbnailUrl: nil, created: nil)
    )
    var searchImagesResult: Result<[ColoringImage], Error> = .success([])

    var fetchImagesCallCount = 0
    var generateImageCallCount = 0
    var searchImagesCallCount = 0
    var lastGeneratePrompt: String?
    var lastSearchQuery: String?

    func setFetchImagesResult(_ result: Result<[ColoringImage], Error>) {
        fetchImagesResult = result
    }

    func setGenerateImageResult(_ result: Result<ColoringImage, Error>) {
        generateImageResult = result
    }

    func setSearchImagesResult(_ result: Result<[ColoringImage], Error>) {
        searchImagesResult = result
    }

    func fetchImages(from baseURL: URL) async throws -> [ColoringImage] {
        fetchImagesCallCount += 1
        return try fetchImagesResult.get()
    }

    func generateImage(prompt: String, baseURL: URL, apiKey: String?) async throws -> ColoringImage {
        generateImageCallCount += 1
        lastGeneratePrompt = prompt
        return try generateImageResult.get()
    }

    func searchImages(query: String, baseURL: URL) async throws -> [ColoringImage] {
        searchImagesCallCount += 1
        lastSearchQuery = query
        return try searchImagesResult.get()
    }
}

File: android/app/src/test/java/com/example/coloringbook/services/MockImageService.kt
package com.example.coloringbook.services

import com.example.coloringbook.TestData
import com.example.coloringbook.models.ColoringImage

class MockImageService : ImageServiceInterface {
    var fetchImagesResult: Result<List<ColoringImage>> = Result.success(emptyList())
    var generateImageResult: Result<ColoringImage> = Result.success(TestData.sampleImage)
    var searchImagesResult: Result<List<ColoringImage>> = Result.success(emptyList())

    var fetchImagesCallCount = 0
    var generateImageCallCount = 0
    var searchImagesCallCount = 0
    var lastGeneratePrompt: String? = null
    var lastSearchQuery: String? = null

    override suspend fun fetchImages(): List<ColoringImage> {
        fetchImagesCallCount++
        return fetchImagesResult.getOrThrow()
    }

    override suspend fun generateImage(prompt: String, appToken: String?): ColoringImage {
        generateImageCallCount++
        lastGeneratePrompt = prompt
        return generateImageResult.getOrThrow()
    }

    override suspend fun searchImages(query: String): List<ColoringImage> {
        searchImagesCallCount++
        lastSearchQuery = query
        return searchImagesResult.getOrThrow()
    }
}
